tools/clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -762,7 +763,18 @@ void clang::EmitBackendOutput(DiagnosticsEngine &Diags,
                               raw_pwrite_stream *OS) {
   EmitAssemblyHelper AsmHelper(Diags, CGOpts, TOpts, LOpts, M);
 
-  AsmHelper.EmitAssembly(Action, OS);
+  // Catch any fatal errors during optimization passes here
```

Comment:
When we add/modify code in existing llvm files, we surround added code blocks with `// HLSL Change Starts` and `// HLSL Change Ends`, or use `// HLSL Change` at end of single line additions.

For this change, this could be simpler by adding the `// HLSL Change` to the end of the `try {` line, surrounding the catch block, and moving the comment into the catch block explaining the reason for the change.